src/wowman/catalog.clj

```diff
@@ -39,15 +43,81 @@
      :total (count addon-list)
      :addon-summary-list addon-list}))
 
+(defn read-catalog
+  "read catalog data from given `catalog-path`"
+  [catalog-path & {:keys [bad-data?]}]
+  ;; todo: cheshire claims to be twice as fast: https://github.com/dakrone/cheshire#speed
+  (if bad-data?
+    (utils/load-json-file-safely catalog-path :bad-data? bad-data?)
+    (utils/load-json-file catalog-path)))
+
 (defn-spec write-catalog ::sp/extant-file
-  "writes catalog to given `output-file` as json. returns path to the output file"
-  [catalog-data ::sp/catalog, output-file ::sp/file]  ;; addon-list at this point has been ordered and doesn't resemble a hashmap anymore
+  "write catalog to given `output-file` as JSON. returns path to the output file"
+  [catalog-data ::sp/catalog, output-file ::sp/file]
   (utils/dump-json-file output-file catalog-data)
   (info "wrote" output-file)
   output-file)
 
+(defn-spec new-catalog ::sp/catalog
+  [addon-list ::sp/addon-summary-list]
+  (let [created (utils/datestamp-now-ymd)
+        updated created]
+    (format-catalog-data addon-list created updated)))
+
+(defn-spec write-empty-catalog! ::sp/extant-file
+  "writes a stub catalog to the given `output-file`"
+  [output-file ::sp/file]
+  (let [created (utils/datestamp-now-ymd)
+        updated created]
+    (write-catalog (new-catalog []) output-file)))
+
 ;;
 
+(defn-spec parse-user-addon (s/or :ok ::sp/addon-summary, :error nil?)
+  "given a string, figures out the source and dispatches to that source logic.
+  supports 'source:source-id' syntax as well as good old fashioned URLs"
```

Comment:
remove "supports 'source:source-id' syntax as well as good old fashioned URLs"

no, it doesn't